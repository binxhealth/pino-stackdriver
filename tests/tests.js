const { join } = require('path')
const { Readable } = require('stream')
const { test } = require('@ianwalter/bff')
const execa = require('execa')

const pinoStackdriver = join(__dirname, '../')
const lineOne = JSON.stringify({
  time: 1544043395681,
  hostname: 'host',
  level: 50,
  msg: 'TestError: blah blah blah',
  pid: 1,
  req: { path: '/test' },
  res: { statusCode: 500 },
  responseTime: 116,
  v: 1
}) + '\n'
const lineTwo = JSON.stringify({
  time: 1544043391245,
  hostname: 'host',
  level: 30,
  msg: 'A user forgot their password',
  pid: 1,
  req: { path: '/api/password-reset' },
  res: { statusCode: 201 },
  responseTime: 288,
  v: 1
}) + '\n'
const lineThree = "{ Error => `Lalala-la lalal-la Elmo's world!` }"

test('pino-stackdriver adds severity to log entry', ({ expect }) => {
  return new Promise(resolve => {
    const stdin = new Readable({ read () {} })
    const cp = execa('node', [pinoStackdriver], { reject: false })

    let lines = []
    cp.stdout.on('data', data => {
      const rawLines = data.toString().split('\n')
      lines = lines.concat(rawLines.filter(line => line))
    })
    cp.stdout.on('close', () => {
      expect(lines).toMatchSnapshot()
      resolve()
    })

    // Write logs to stream.
    stdin.pipe(cp.stdin)
    stdin.push(lineOne)
    stdin.push(lineTwo)
    stdin.push(lineThree)

    // Push null to close the streams.
    stdin.push(null)
    cp.stdin.push(null)
  })
})
