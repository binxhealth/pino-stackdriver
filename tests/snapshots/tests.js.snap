// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pino-stackdriver adds severity to log entry 1`] = `
Array [
  "{\\"time\\":\\"2018-12-05T20:56:35.681Z\\",\\"hostname\\":\\"host\\",\\"level\\":50,\\"msg\\":\\"TestError: blah blah blah\\",\\"pid\\":1,\\"req\\":{\\"path\\":\\"/test\\"},\\"res\\":{\\"statusCode\\":500},\\"responseTime\\":116,\\"v\\":1,\\"severity\\":\\"ERROR\\"}",
  "{\\"time\\":\\"2018-12-05T20:56:31.245Z\\",\\"hostname\\":\\"host\\",\\"level\\":30,\\"msg\\":\\"A user forgot their password\\",\\"pid\\":1,\\"req\\":{\\"path\\":\\"/api/password-reset\\"},\\"res\\":{\\"statusCode\\":201},\\"responseTime\\":288,\\"v\\":1,\\"severity\\":\\"INFO\\"}",
  "{ Error => \`Lalala-la lalal-la Elmo's world!\` }",
]
`;

exports[`pino-stackdriver adds severity to log entry 2`] = `
Object {
  "hostname": "host",
  "level": 30,
  "msg": "A user forgot their password",
  "pid": 1,
  "req": Object {
    "path": "/api/password-reset",
  },
  "res": Object {
    "statusCode": 201,
  },
  "responseTime": 288,
  "severity": "INFO",
  "time": "2018-12-05T20:56:31.245Z",
  "v": 1,
}
`;
